ros_buildfarm/aggregate_status.py

```diff
@@ -0,0 +1,291 @@
+import os
+import re
+import yaml
+import requests
+import collections
+from .config import get_index as get_config_index, get_release_build_files
+
+"""
+
+  Aggregate Status takes the yaml generated by the different CPU architectures (e.g. ros_lunar_default, ros_lunar_ds)
+  and combines them into one yaml file per ROS distro. It also creates an English description of each package across
+  all of the architectures. See examples below.
+
+  Naming Conventions
+   * distro refers to ROS Distro (e.g. indigo)
+   * machine refers to different release builds (e.g. default, uxhf, dsv8)
+   * os_name refers to the name of the Operating System (e.g. ubuntu, debian)
+   * os_flavor refers to specific distros of the os (e.g. xenial, jessie)
+   * cpu refers to the cpu architecture (or source), (e.g. i386, amd64, source)
+   * candidate refers to different candidate builds, (e.g. build, test, main)
+   * combo refers to the os_flavor + the cpu
+
+  Example Descriptions:
+   - released - All candidates have the same version number
+   - waiting for new release - Build/test have the same version number, but there's nothing for main
+   - waiting for re-release - Build/test have the same version number, but there's an older version for main
+   - waiting for new/re-release - Build/test have the same version number, but there's either nothing or an older
+                                  version for main
+   - source builds, binary doesn't - If the source builds, but all the other cpus do not
+   - does not build on X - If the source and some of the cpus build, but some do not
+"""
+
+CANDIDATES = ['build', 'test', 'main']  # Not imported from status_page to avoid potential circular dependency
+
+YAML_FOLDER = 'http://repositories.ros.org/status_page/yaml/'
```

Comment:
I noted this in the [original PR](https://github.com/ros-infrastructure/ros_buildfarm/pull/555). I could not find anywhere in the [current production config](https://github.com/ros-infrastructure/ros_buildfarm_config/tree/production) that matched `'http://repositories.ros.org'` other than [some longer urls](https://github.com/ros-infrastructure/ros_buildfarm_config/blob/production/index.yaml#L85), but maybe I missed something. Where I can I get/derive this URL.

Reply:
The `prerequisites` is not the right information for this.

The generated files are being published via SSH and the only location where that target is specified atm is in the `buildfarm_deployment_config` afaik.

Reply:
How would you recommend I tackle this then? Is there a way for me to grab them via ssh?

Reply:
Where ever these status pages are being configured there needs to be a config file (as for all the other job types) where this URL can be set. That part of the logic might not be in this PR as far as I can see since nothing uses the new Python module yet.